HW1/P2/P2b.py

```diff
@@ -1,3 +1,25 @@
 from P2 import *
+from pyspark import SparkContext
+import random
 
-# Your code here
+# Set up partitions and context
+N = 2000
+sc = SparkContext("local", "P2b")
+
+# Partition a 10x10 cartesian grid of workers
+# Same as P2a, except the ordering is randomized
+rdd = sc.parallelize(sorted(range(N), key=lambda k: random.random()), 10)
```

Comment:
Partition using random could be a better solution, however, in order to be more targeted and efficient, customise partition strategy according to pixels distribution on the graph using spark "partitioner" could be an optimised solution.